custom/icds_reports/management/commands/fetch_form_data.py

```diff
@@ -0,0 +1,137 @@
+import csv
+from datetime import date
+
+from dateutil.relativedelta import relativedelta
+from django.core.management.base import BaseCommand
+from django.db import connections, transaction
+
+from corehq.apps.userreports.util import get_table_name
+from custom.icds_reports.models import AwcLocation
+from custom.icds_reports.utils.connections import get_icds_ucr_citus_db_alias
+
+query = """
+    SELECT
+        state_id, count(*) as form_count
+        FROM "{table_name}"
+        WHERE inserted_at>='{start_date}'::date AND inserted_at<'{end_date}'::date GROUP BY state_id
+"""
+
+query_usage = """
+    SELECT
+        awc.state_id, count(*) as form_count
+        FROM "{table_name}" use
+        LEFT JOIN awc_location awc ON (
+            awc.doc_id = use.awc_id AND awc.aggregation_level=5
+        )
+        WHERE use.inserted_at>='{start_date}'::date AND use.inserted_at<'{end_date}'::date AND 
+        {usage_field_name}=1 AND awc.aggregation_level=5 GROUP BY awc.state_id
+"""
+
+usage_forms = {
+    "Register Household": "add_household",
+    "Add Member": "add_person",
+    "Post Natal Care": "pnc",
+    "Exclusive Breast Feeding": "ebf",
+    "Growth Monitoring": "gmp"
+}
+
+forms = {
+    "Register Household": "static-usage_forms",
+    "Add Member": "static-usage_forms",
+    "Add pregnancy": "static-dashboard_add_pregnancy_form",
+    "Availing services": "static-availing_service_form",
+    "Daily feeding": "static-daily_feeding_forms",
+    "Birth Preparedness": "static-dashboard_birth_preparedness_forms",
+    "Delivery": "static-child_delivery_forms",
+    "Post Natal Care": "static-usage_forms",
+    "Exclusive Breast Feeding": "static-usage_forms",
+    "Complementary Feeding": "static-complementary_feeding_forms",
+    "Growth Monitoring": "static-usage_forms",
+    "Additional Growth Monitoring": "static-dashboard_growth_monitoring_forms",  # special case
+    "THR Distribution": "static-dashboard_thr_forms",
+    "Infratructure Details": "static-infrastructure_form_v2",
+    "VHSND Survey": "static-vhnd_form",
+    "Visitor Register": "static-visitorbook_forms",
+    "AWC Visit Form": "static-awc_mgt_forms",
+    "Beneficiary Feedback": "static-ls_home_visit_forms_filled",
+    "VHSND observation Form": "static-ls_vhnd_form",
+}
+
+dates = [
```

Comment:
what are these dates

Reply:
These are the start dates of the weeks for which we have to fetch data. Although move them to loop instead of a static list.